### DISCUSSION TITLE
Why we need pass the record object into page in resource edit action?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1885] [Feature requests & Ideas] [answered]
As added in the following line.

https://github.com/luckyframework/avram/blob/main/src/lucky/tasks/gen/templates/resource/actions/edit.cr.ecr#L6

I ask this question just because curious, because i consider this is not necessary, we can always get the original object from the operation object use the `operation.record`, right?

**Comment by jwoertink** [comment, marked as answer]
You can get it from `operation.record`, but that is nilable so you'd have to call `operation.record.not_nil!` or cast it ever time. We pass the resource in to guarantee we have the value, and the compiler knows the type.
